### PR TITLE
feat: support fully customized toasts #53

### LINKED ISSUE #53: auro-toast: Flexibility to choose icon, background color, and not have it auto-dismiss
### Is your feature request related to a problem? Please describe.

I was recommended by auro designers to use toast for a dismissible popup containing minimal text, that the guest would have to dismiss themselves. But the current version doesn't meet all my needs, nor do any of the other components available to me.

Example:
image

### Describe the solution you'd like

I would like to be able to make a personalized toast, outside of the current success/default/warning/info options, where we could select any auro icon (or subset of auro icons) along with using any of our container/background color tokens. The ability to have it auto-dismiss, or not, would also be beneficial.

### Describe alternatives you've considered

Auro dialog specs are too large for my use case, and popup only appears on hover/click. We want this there upon landing on a page.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Add support for fully customized toast notifications, enabling custom SVG icons, color customization, and the ability to disable auto-hide. Update documentation to reflect these new customization options.

New Features:
- Introduce support for fully customized toast notifications, allowing users to pass custom SVGs, recolor notifications, and disable auto-hide behavior.

Enhancements:
- Refactor toast component to handle custom SVG icons and improve the handling of toast visibility and auto-hide functionality.

Documentation:
- Update documentation to include instructions on customizing toast notifications, including using custom SVGs, recoloring, and disabling auto-hide.

### CODE REVIEW

**Comment by jason-capsule42** [comment]
@sourcery-ai review

**Comment by jason-capsule42** [comment]
@sourcery-ai review

File: src/auro-toast.js [hunk 1]
```diff
@@ -131,6 +132,10 @@ export class AuroToast extends LitElement {
       },
       noIcon: {
         type: Boolean
+      },
+      disableAutoHide: {
```
Comment (sun-mota) [inline src/auro-toast.js:138]: default value missing in constructor; for all properties in this component.

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Running a `build:release` causes the `index.min.js` and `api.min.js` causes the `initExamples` function export to be removed, therefore breaking the examples on the demo pages.

**Comment by jason-capsule42** [comment]
> Running a `build:release` causes the `index.min.js` and `api.min.js` causes the `initExamples` function export to be removed, therefore breaking the examples on the demo pages.

This has been resolved.

File: src/auro-toast.js (hunk 1 above)
Comment (jason-capsule42) [reply src/auro-toast.js:138]: We don't want defaults on this properties (even `false`) on this attributes. Having a value of any kind will cause the reflected attributes to appear in the DOM and force style and functional behavior that shouldn't occur until after a value is set. (e.g. applying visible=false will put that into the DOM and the presence of the visible attribute at all forces the toast to display.)

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Two issues:

`X` icon does not look vertically centered when the notification icon is present
Screen Shot 2024-11-08 at 9 01 11 AM

This error comes up when clicking the `X` button on new custom toast example
Screen Shot 2024-11-08 at 9 01 35 AM

**Review by jordanjones243 — DISMISSED** [review]

**Review by jordanjones243 — DISMISSED** [review]

**Pull request merged**